compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -192,18 +193,20 @@ object MainGenericRunner {
 
       case ExecuteMode.PossibleRun =>
         val newClasspath = (settings.classPath :+ ".").flatMap(_.split(classpathSeparator).filter(_.nonEmpty)).map(File(_).toURI.toURL)
-        import dotty.tools.runner.RichClassLoader._
-        val newClassLoader = ScalaClassLoader.fromURLsParallelCapable(newClasspath)
+        import dotty.tools.runner.RichClassLoader.{getClass as _, *}
+        val newClassLoader = ScalaClassLoader.fromURLsParallelCapable(newClasspath, getClass.getClassLoader)
```

Comment:
This change does not seem to be needed to make `tests/run-with-compiler/i14541.scala` pass. Why was this added? How could I test this?

Reply:
This is the 2nd commit about whether to do `-usejavacp` behavior from Scala 2, i.e., use the class loader you were given. I'm doubtful both ways.

Reply:
If that is the case, I would remove this change and the second test. The third test already covers the fix. We could have a look at this cp change in a separate PR as we don't know or have time to test unintended consequences of this change. I would prefer to only fix the regression in this PR as this fill be back ported. We also want the backport quickly to be able to release 3.1.2 as soon as possible.

Reply:
@som-snytt we also have a -usejavacp flag checked in some of our she'll scripts I believe, shouldn't we check it here too?